datamad2/views.py

```diff
@@ -306,6 +314,39 @@ class MyAccountDetailsView(LoginRequiredMixin, TemplateView):
     template_name = 'datamad2/user_account/my_account.html'
 
 
+class MyAccountNewUserView(LoginRequiredMixin, UserPassesTestMixin, CreateView):
+    template_name = 'datamad2/user_account/datacentre_new_users.html'
+    model = User
+    form_class = UserForm
+
+    def test_func(self):
+        return self.request.user.is_admin
+
+    def get_success_url(self):
+        messages.success(self.request, 'User added successfully')
+        return reverse('new_user')
+
+    def get_object(self, **kwargs):
```

Comment:
I am not sure you will need the get object method as you are using a create view.

Reply:
Oh yeah! I should have read your description